File: leetcode/go/problem91.go
	func numDecodings(s string) int {
		a := 1
		b := 0
		if s[0] != '0' {
			b = 1
		}
		for i := 2; i <= len(s); i++ {
			cur := 0
			d, _ := strconv.Atoi(s[i-1 : i])
			if d >= 1 && d <= 9 {
				cur += b
			}

			dd, _ := strconv.Atoi(s[i-2 : i])
			if dd >= 10 && dd <= 26 {
				cur += a
			}
			a, b = b, cur
		}
		return b
	}

File: leetcode/python/problem91.py
class Solution:
    def numDecodings(self, s: str) -> int:
        if s[0] == "0":
            return 0
        dp = [0 for _ in s]
        dp[0] = 1
        for i in range(1, len(s)):
            r = 0
            if 10 <= int(s[i - 1 : i + 1]) <= 26:
                if i > 1:
                    dp[i] += dp[i - 2]
                else:
                    dp[i] += 1
            if 1 <= int(s[i]) <= 9:
                dp[i] += dp[i - 1]
        return dp[-1]
